packages/block-editor/src/components/rich-text/index.js

```diff
@@ -2,1112 +2,128 @@
-								onCompositionEnd={ this.onCompositionEnd }
-								onKeyDown={ this.onKeyDown }
-								onFocus={ this.onFocus }
-								onBlur={ this.onBlur }
-								onMouseDown={ this.onPointerDown }
-								onTouchStart={ this.onPointerDown }
-								setRef={ this.setRef }
-							/>
-							{ isPlaceholderVisible &&
-								<Tagname
-									className={ classnames( 'editor-rich-text__editable block-editor-rich-text__editable', className ) }
-									style={ style }
-								>
-									{ MultilineTag ? <MultilineTag>{ placeholder }</MultilineTag> : placeholder }
-								</Tagname>
-							}
-							{ isSelected && <FormatEdit value={ record } onChange={ this.onChange } /> }
-						</>
+const wrapperClasses = 'editor-rich-text block-editor-rich-text';
+const classes = 'editor-rich-text__editable block-editor-rich-text__editable';
+
+function RichTextWraper( {
+	tagName,
+	value: originalValue,
+	onChange: originalOnChange,
+	selectionStart,
+	selectionEnd,
+	onSelectionChange,
+	multiline,
+	onTagNameChange,
+	inlineToolbar,
+	wrapperClassName,
+	className,
+	autocompleters,
+	onReplace,
+	onRemove,
+	onMerge,
+	onSplit,
+	isCaretWithinFormattedText,
+	onEnterFormattedText,
+	onExitFormattedText,
+	canUserUseUnfilteredHTML,
+	isSelected: originalIsSelected,
+	onCreateUndoLevel,
+	placeholder,
+	keepPlaceholderOnFocus,
+	// From experimental filter.
+	...experimentalProps
+} ) {
+	let adjustedValue = originalValue;
+	let adjustedOnChange = originalOnChange;
+
+	// Handle deprecated format.
+	if ( Array.isArray( originalValue ) ) {
+		adjustedValue = children.toHTML( originalValue );
+		adjustedOnChange = ( newValue ) => originalOnChange( children.fromDOM(
+			__unstableCreateElement( document, newValue ).childNodes
+		) );
+	}
+
+	return (
+		<RichText
+			{ ...experimentalProps }
+			value={ adjustedValue }
+			onChange={ adjustedOnChange }
+			selectionStart={ selectionStart }
+			selectionEnd={ selectionEnd }
+			onSelectionChange={ onSelectionChange }
+			tagName={ tagName }
+			wrapperClassName={ classnames( wrapperClasses, wrapperClassName ) }
+			className={ classnames( classes, className ) }
+			placeholder={ placeholder }
+			keepPlaceholderOnFocus={ keepPlaceholderOnFocus }
+			__unstableIsSelected={ originalIsSelected }
+			__unstablePatterns={ getPatterns() }
+			__unstableEnterPatterns={ getEnterPatterns() }
+			__unstablePasteHandler={ pasteHandler }
+			__unstableAutocomplete={ Autocomplete }
+			__unstableAutocompleters={ autocompleters }
+			__unstableOnReplace={ onReplace }
+			__unstableOnRemove={ onRemove }
+			__unstableOnMerge={ onMerge }
+			__unstableOnSplit={ onSplit }
+			__unstableMultiline={ multiline }
+			__unstableIsCaretWithinFormattedText={ isCaretWithinFormattedText }
+			__unstableOnEnterFormattedText={ onEnterFormattedText }
+			__unstableOnExitFormattedText={ onExitFormattedText }
+			__unstableCanUserUseUnfilteredHTML={ canUserUseUnfilteredHTML }
+			__unstableOnCreateUndoLevel={ onCreateUndoLevel }
```

Comment:
Seems like a lot of unstable Props :) How do we plan to solve these in the future?

Reply:
Yes! :) But I don't want to rewrite it all at once so I temporarily marked them unstable.